core/server/api/v2/settings.js

```diff
@@ -92,6 +92,12 @@ module.exports = {
                             message: common.i18n.t('errors.api.settings.accessCoreSettingFromExtReq')
                         }));
                     }
+
+                    if (setting.key === 'image_sizes' && !(frame.options.context && frame.options.context.internal)) {
```

Comment:
Good one! Think it's pretty safe to rely on having options.context at this point as it's always there - https://github.com/TryGhost/Ghost/blob/master/core/server/api/shared/pipeline.js#L208

Reply:
Although found [this comment](https://github.com/TryGhost/Ghost/blob/master/core/server/api/v2/settings.js#L23) in the codebase:
> // CASE: no context passed (functional call)

Haven't found any direct usages of `.query(` on settings controller though :man_shrugging: I think I'll leave the context check just in case with a comment for a later cleanup/headsup...

Reply:
:confused: 	damn this condition combination `(frame.options.context && frame.options.context.internal)` is used only in settings controller but don't see any clear reason for it. will put it as a last piece to check in the TODO list